### PR TITLE
Update to follow modern/secure electron convention

### PR DESCRIPTION
**Pull request by mpatankar6**
Moves all ipcrenderer calls to a preload script, which aligns with best
electron practices more so students can understand how to properly
structure an electron program rather than the security benefit.

Updates electron to v20.0.2

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 Object.defineProperty(exports, '__esModule', { value: true });
 
+const path = require("path");
```
Comment (ErikBoesen) [inline src/main.js:4]: Use single quotes for consistency

File: src/main.js [hunk 2]
```diff
@@ -103,7 +104,7 @@ function createWindow() {
         show: false,
         icon: __dirname + '/../images/icon.png',
         webPreferences: {
-            nodeIntegration: true
+            preload: path.join(__dirname, "preload.js"),
```
Comment (ErikBoesen) [inline src/main.js:107]: Here and other places as well

File: src/ui.js [hunk 3]
```diff
@@ -107,5 +107,5 @@ ui.armPosition.oninput = function() {
 };
 
 addEventListener('error',(ev)=>{
-    ipc.send('windowError',{mesg:ev.message,file:ev.filename,lineNumber:ev.lineno})
+    window.api.sendWindowError({mesg:ev.message,file:ev.filename,lineNumber:ev.lineno})
```
Comment (ErikBoesen) [inline src/ui.js:110]: Since you're here, could you fix this spacing?

**Review by ErikBoesen — CHANGES_REQUESTED** [review]
Just tiny styling tweaks. Thank you for the contribution!

**Comment by mpatankar6** [comment]
Should be all set now

**Review by ErikBoesen — APPROVED** [review]

**Pull request merged**